app/code/Magento/CatalogUrlRewrite/Observer/ProcessUrlRewriteOnChangeProductVisibilityObserver.php

```diff
@@ -15,6 +16,8 @@
 
 /**
  * Consider URL rewrites on change product visibility via mass action
+ *
+ * @deprecated Url rewrites will be created for products with all visibility types.
```

Comment:
I think this comment isn't clear that this observer will be never called. Also not clear where url rewrites will be created. I thing we need to add something like this:
```suggestion
 * @deprecated Class is not used anymore. Url rewrites will be created for products with all visibility types.
 * @see \Magento\CatalogUrlRewrite\Observer\ProductProcessUrlRewriteSavingObserver
```